odl/discr/partition.py

```diff
@@ -546,6 +555,139 @@ def squeeze(self):
         newset = self.set[self.grid.inondeg]
         return RectPartition(newset, self.grid.squeeze())
 
+    def index(self, value, floating=False):
+        """Return the index of a value in the domain.
+
+        Parameters
+        ----------
+        value : ``self.set`` element
+            The point to find the index of.
+        floating : bool
+            True if the index should also give the position inside the voxel.
+            This is given by returning the integer valued index of the voxel
+            plus the partial distance from the left to the right boundary.
+
+        Returns
+        -------
+        index : int/float or tuple of int/float
+            Index of the value, as counted from the left.
+            If ``self.ndim>1`` the result is a tuple, else a scalar.
+            If ``floating=True`` the scalar is a float, else an int.
+
+        Examples
+        --------
+        Get the indices of start and end:
+
+        >>> p = uniform_partition(0, 2, 5)
+        >>> p.index(0)
+        0
+        >>> p.index(2)
+        4
+
+        Also supports points inside voxels, in that case returning the index
+        of the voxel:
```

Comment:
For points inside voxels, the index of the containing cell is returned:

would be my favorite